xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -1231,14 +1230,15 @@ public String toString() {
           return StructOrError.fromError(
               "Unknown cluster specifier: " + proto.getClusterSpecifierCase());
       }
-      long timeoutNano = TimeUnit.SECONDS.toNanos(15L);  // default 15s
-      if (proto.hasMaxGrpcTimeout()) {
-        timeoutNano = Durations.toNanos(proto.getMaxGrpcTimeout());
-      } else if (proto.hasTimeout()) {
-        timeoutNano = Durations.toNanos(proto.getTimeout());
-      }
-      if (timeoutNano == 0) {
-        timeoutNano = Long.MAX_VALUE;
+      long timeoutNano = 0L;
```

Comment:
Nope, if you look at the table in the gRFC, a value of `0` for those fields is effectively equivalent to it is not set.

Reply:
The table is after applying the fallback if possible.
_"Note that the max_stream_duration column refers to the effective setting based on both RouteAction.max_stream_duration.max_stream_duration and the max_stream_duration from the HTTP Connection Manager's common_http_options."_

Reply:
See my updated comment. I don't understand your point. But, a value of `0` is effectively equivalent to it is not set, do you agree with that?

Reply:
Case 1 corresponds to row 4 of the table.
> If maxStreamDuration.hasGrpcTimeoutHeaderMax() and maxStreamDuration.getGrpcTimeoutHeaderMax() != 0: use it. Otherwise, go to next step.

This is completely broken for case 1.

Reply:
Ah, that table is problematic. If you interpret it in that way, the second last row is a counter-example...

Reply:
> the second last row is a counter-example

The second from the last row does not conflict with any other row, it just takes min(20, infinite).

Reply:
Okay, fixed. Falling back to `max_stream_duration` from the `HttpConnectionManager.common_http_options` will only take place if both `!maxStreamDuration.hasGrpcTimeoutHeaderMax()` and `!maxStreamDuration.hasMaxStreamDuration()` hold. We don't need to make the last fallback value `nullable` as its value of `0` and unset are treated equivalently (aka, as infinite) when it is indeed used.